src/_base/harness/attributes/common.yml

```diff
@@ -297,6 +297,21 @@ attributes.default:
       steps: []
 
   console:
+    healthcheck:
+      # we want that enabled only for local builds
+      enabled: false
+      action:
+        steps:
+          - echo 'Entrypoint tasks completed' > /tmp/HEALTHCHECK.txt
```

Comment:
do we definitely have use cases we can think of other than for end state indicators like this?

Reply:
This was the most basic check I could think of, as explained in the description of this PR
I am happy to implement a more precise and truthful test of health.
This is a basic example that works. just it.

Reply:
note this isn't the health check itself but the trigger that indicates its started

Reply:
yes the `action` is meant to generate a measurable state.
In this case we are considering `console` "healthy" when all the `usermod` and `groupmod` commands that are inside the dynamic entrypoint to be complete

Adding a function that is executed after those commands and that creates a `semaphore` file was the most accurae way to create a measurable think to say "oh, it's healthy"
of course, the `healthcheck_action` is customisable and also can be left doing nothing
and we simply can customise the `healthcheck_test` to check anything else we deem as a health parameter 

I see `healthcheck_action` as a hook where we can do something custom to generate a health factor IF we need to. and now we needed to (I couldn't think about anything smarter)

Reply:
`I am happy to implement a more precise and truthful test of health.` yes but testing the health is the healthcheck's responsibility, rather than the entrypoint. Here is just a startup check, for a console which has no application running, so I can't see there need for anything other than indicating the process has completed.

Reply:
point taken!
what if I move `healthcheck_action()` to `entrypoint.sh.twig` and action it only if the environment is dynamic?

Reply:
> Here is just a startup indicator, for a console which has no application running,
yes
> so I can't see there need for anything other than indicating the process has completed.
true, for now thing more seems needed
I left it customisable just in case we might need it in the future

Reply:
I think the realm of the scripting of this would unlikely go beyond a startup indicator file, so could be hard coded in the entrypoint.sh.twig even, regardless of dynamic/static modes

Reply:
and entrypoint.steps would then cater to anything that we can't think of yet.

Reply:
> and console.entrypoint.steps would then cater to anything that we can't think of yet.
you mean that `entrypoint.steps` could do what the `healthcheck.action.steps` are meant to do now?

if that case I agree that we can
1. hardcode the `healthcheck_action` function and steps in `entrypoint.sh.twig`
2. if a project needs a custom indicator of the `console` health rather than `/HEALTCHECK.txt` they can add those steps into `entrypoint.steps`
3. and modify `healthcheck.test.steps` or `healthcheck.test.command` to look for the custom indicator

**Let me know if my interpretation of your comment is correct, and I will implement it**

Reply:
yes, though I think something that more simply describes the state as filename would be better than `healthcheck.txt`, e.g. `/run/started`

Reply:
moved the health check "checkpoint" into an embedded function in "entrypoint.sh.twig"